NYT.py

```diff
@@ -0,0 +1,213 @@
+os.environ['KAGGLE_CONFIG_DIR'] = '/kaggle'
+
+!kaggle datasets download -d aryansingh0909/nyt-articles-21m-2000-present --force
+!unzip nyt-articles-21m-2000-present.zip
+
+dbutils.fs.mv("file:/databricks/driver/nyt-metadata.csv", "/FileStore/nyt-metadata.csv")
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ###Reading .csv file to DataFrame
+
+# COMMAND ----------
+
+from pyspark.sql.types import StructField, StructType, StringType, TimestampType, IntegerType, FloatType
+
+nyt_schema = StructType([
+    StructField('abstract', StringType(), True), 
+    StructField('web_url', StringType(), False), 
+    StructField('snippet', StringType(), True), 
+    StructField('lead_paragraph', StringType(), True), 
+    StructField('print_section', StringType(), True), 
+    StructField('print_page', FloatType(), True), 
+    StructField('source', StringType(), False), 
+    StructField('multimedia', StringType(), True), 
+    StructField('headline', StringType()), 
+    StructField('keywords', StringType(), True),
+    StructField('pub_date', TimestampType(), False), 
+    StructField('document_type', StringType(), True), 
+    StructField('news_desk', StringType(), True), 
+    StructField('section_name', StringType(), True), 
+    StructField('byline', StringType(), True),
+    StructField('type_of_material', StringType(), True), 
+    StructField('_id', StringType(), False), 
+    StructField('word_count', FloatType(), True), 
+    StructField('uri', StringType(), False), 
+    StructField('subsection_name', StringType(), True),
+    StructField('corrupted', StringType(), True)
+])
+
+nyt_articles = (spark
+    .read
+    .option('header', 'true')
+    .option("quote", "\"")
+    .option("escape", "\"")
+    .option("timestampFormat", "yyyy-MM-dd HH:mm:ss") \
+    .schema(nyt_schema)
+    .csv('/FileStore/nyt-metadata.csv', multiLine=True))
+
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ###Showing the DataFrame
+
+# COMMAND ----------
+
+from pyspark.sql.functions import rand 
+display(nyt_articles.orderBy(rand()).limit(10))
+nyt_articles.printSchema()
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ### Repairing data
+
+# COMMAND ----------
+
+from pyspark.sql.functions import col, to_timestamp, levenshtein
+
+#there is a row that is split into two with null values 
+row1 = nyt_articles.filter(col("web_url") == "https://www.nytimes.com/2022/04/28/sports/football/nfl-draft-hats-2022.html")
+row2 = nyt_articles.filter(col("section_name") == "nyt://article/2aec1dad-e9d0-5a3c-931b-cc93890bb60e")
+
+row1 = row1.select(nyt_articles.columns[:3])
+row2 = row2.select(nyt_articles.columns[:18])
+combined_row = row1.crossJoin(row2)
+combined_row = combined_row.toDF(*nyt_articles.columns)
+display(combined_row)
+
+# COMMAND ----------
+
+#replace 2 broken rows with repaired one
+nyt_articles = nyt_articles.filter(col("web_url") != "https://www.nytimes.com/2022/04/28/sports/football/nfl-draft-hats-2022.html")
```

Comment:
If you have a repeating url - move it to a variable. This way it would be easier to change it